src/data-interfaces/HoloFuelDnaInterface.js

```diff
@@ -260,34 +260,36 @@ const HoloFuelDnaInterface = {
 
       const { transactions } = await createZomeCall('transactions/list_transactions')(params)
       const nonActionableTransactions = transactions.map(presentTransaction).filter(tx => !(tx instanceof Error))
-      const noDuplicateIds = _.uniqBy(nonActionableTransactions, 'id')
-      const presentedCompletedTransactions = await getTxWithCounterparties(noDuplicateIds.filter(tx => tx.status === 'completed'))
+      const uniqueNonActionableTransactions = _.uniqBy(nonActionableTransactions, 'id')
+      const presentedCompletedTransactions = await getTxWithCounterparties(uniqueNonActionableTransactions.filter(tx => tx.status === 'completed'))
       return presentedCompletedTransactions.sort((a, b) => a.timestamp > b.timestamp ? -1 : 1)
     },
     allActionable: async () => {
       const { requests, promises, declined, canceled } = await createZomeCall('transactions/list_pending')()
       const actionableTransactions = await requests.map(r => presentPendingRequest(r)).concat(promises.map(p => presentPendingOffer(p))).concat(declined.map(presentDeclinedTransaction)).concat(canceled.map(presentIncomingCanceledTransaction))
-      const presentedActionableTransactions = await getTxWithCounterparties(actionableTransactions)
+      const uniqActionableTransactions = _.uniqBy(actionableTransactions, 'id')
+      const presentedActionableTransactions = await getTxWithCounterparties(uniqActionableTransactions)
 
       return presentedActionableTransactions.sort((a, b) => a.timestamp > b.timestamp ? -1 : 1)
     },
     allWaiting: async () => {
       const { transactions } = await createZomeCall('transactions/list_transactions')()
       const nonActionableTransactions = transactions.map(presentTransaction).filter(tx => !(tx instanceof Error))
       /* NOTE: Filtering out duplicate IDs should prevent an already completed tranaction from displaying as a pending tranaction if any lag occurs in data update layer.  */
-      const noDuplicateIdsWaitingList = _.uniqBy(nonActionableTransactions, 'id')
-      const transactionIds = await HoloFuelDnaInterface.transactions.allDeclinedTransactions()
+      const uniqueNonActionableTransactions = _.uniqBy(nonActionableTransactions, 'id')
```

Comment:
I like the change in naming convention to 'unique<TRANSACTION_TYPE>`, but let's call this `uniqueWaitingTransactions` as this is for the waitingTransactions endpoint. - WaitingTransactions is still different than nonActionationableTransactions, as nonActionable includes all declined (and cancelled - when that was a thing).